### DISCUSSION TITLE
Listening for data changes within Firestore

### DISCUSSION THREAD

**Discussion by giancoppola** [GodotNuts/GodotFirebase#309] [Q&A] [answered]
Hi,

I've been through the Wiki and I can't seem to figure this one out, I can authenticate and have users write content to my Firestore database, but I'm struggling to find a way to listen for updates to the data in realtime for Firestore, is there something similar to the new_data_update signal for the Realtime Database?

Thanks,
Gian

**Comment by WolfgangSenff** [comment, marked as answer]
Unfortunately not! The method the Firebase team implemented to listen for live updates in Firestore chose to implement does not have a REST-based implementation. This means we can't implement it in our plugin. Sorry! (It would require an entire implementation of gRPC, something none of us is capable of doing.)

**Reply by giancoppola** [reply]
Thanks for replying so quickly and for the clarification! Good to know, I'll stick to Realtime Database instead in that case

**Reply by WolfgangSenff** [reply]
No problem! And you can technically use both. It's a little bit resource intensive, but if it makes sense, you can do notifications through RTDB that tell you when to check Firestore for updates. It's only really effective for things that aren't updated very often though, tbh - Firestore can be expensive to make a ton of requests to, so anything that requires even moderately fast updates, it is painful.
